ov/labels.go

```diff
@@ -0,0 +1,237 @@
+/*
+(c) Copyright [2021] Hewlett Packard Enterprise Development LP
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ov
+
+import (
+	"encoding/json"
+	"github.com/HewlettPackard/oneview-golang/rest"
+	"github.com/HewlettPackard/oneview-golang/utils"
+	"github.com/docker/machine/libmachine/log"
+)
+
+type AllLabels struct {
+	Category    string        `json:"category,omitempty"`
+	Count       int           `json:"count,omitempty"`
+	Created     string        `json:"created,omitempty"`
+	ETAG        utils.Nstring `json:"eTag,omitempty"`
+	Members     []Member      `json:"members,omitempty"`
+	Modified    string        `json:"modified,omitempty"`
+	NextPageUri utils.Nstring `json:"nextPageUri,omitempty"`
+	PrevPageUri utils.Nstring `json:"prevPageUri,omitempty"`
+	Start       int           `json:"start,omitempty"`
+	Total       int           `json:"total,omitempty"`
+	Type        string        `json:"type,omitempty"`
+	Uri         utils.Nstring `json:"uri,omitempty"`
+}
+
+type Member struct {
+	Category string        `json:"category,omitempty"`
+	Created  string        `json:"created,omitempty"`
+	ETAG     string        `json:"eTag,omitempty"`
+	Modified string        `json:"modified,omitempty"`
+	Name     string        `json:"name,omitempty"`
+	Type     string        `json:"type,omitempty"`
+	Uri      utils.Nstring `json:"uri,omitempty"`
+}
+
+type NewLabels struct {
+	LabelNames  []string      `json:"labels,omitempty"`
+	ResourceUri utils.Nstring `json:"resourceUri,omitempty"`
+}
+
+type AssignedLabel struct {
+	Category    string        `json:"category,omitempty"`
+	Created     string        `json:"created,omitempty"`
+	ETAG        utils.Nstring `json:"eTag,omitempty"`
+	Labels      []Label       `json:"labels,omitempty"`
+	Modified    string        `json:"modified,omitempty"`
+	ResourceUri utils.Nstring `json:"resourceUri,omitempty"`
+	Type        string        `json:"type,omitempty"`
+	Uri         utils.Nstring `json:"uri,omitempty"`
+}
+
+type Label struct {
```

Comment:
Whats the use of this when NewLabels is being used?

Reply:
It was used to create new labels, but I think we can used AssignedLabel struct to create new ones, hence removed it now.